plutus-core/plutus-ir/src/PlutusIR/Compiler.hs

```diff
@@ -90,7 +91,10 @@ compileToReadable =
 -- Note: the result *does* have globally unique names.
 compileReadableToPlc :: Compiling m e uni fun a => Term TyName Name uni fun (Provenance a) -> m (PLCTerm uni fun a)
 compileReadableToPlc =
-    NonStrict.compileNonStrictBindings
+    -- See Note [Marking]. Failing to call 'markNonFreshTerm' results in variable capture
+    -- even in simple cases causing tests to fail.
+    through markNonFreshTerm
```

Comment:
I don't see why putting this here changes anything. This gets run after `compileToReadable`, which starts with `rename`, which given your other change should get marking right. It's weird that we could somehow get here and have problems...

Reply:
Oh, indeed. I have a hypothesis though: both `inline` and `removeDeadBindings` call `runQuoteT` and so we lose the information of what was the unused unique and start from the same place where we were before calling `inline` and `removeDeadBindings`, which is how we get variable capture. I'll try dropping `runQuoteT` from both of these functions.

Reply:
Yeah, that sounds right. They probably shouldn't do that.

Reply:
I've dropped those `runQuoteT`s, rereview.